services/datamanager/builtin/builtin.go

```diff
@@ -375,13 +376,13 @@ func (svc *builtIn) syncDataCaptureFiles() error {
 		oldFiles = append(oldFiles, collector.Collector.GetTarget().Name())
 		collector.Collector.SetTarget(nextTarget)
 	}
-	svc.lock.Unlock()
 	svc.syncer.Sync(oldFiles)
 	return nil
 }
 
 func (svc *builtIn) buildAdditionalSyncPaths() []string {
 	svc.lock.Lock()
+	defer svc.lock.Unlock()
```

Comment:
This one I actually don't think had any deadlock risk, but still a good best practice to defer so added it here